modules/server/src/main/scala/seqexec/server/nifs/NifsControllerEpics.scala

```diff
@@ -271,25 +271,27 @@ object NifsControllerEpics extends NifsEncoders {
         // * The the current offset is not  0 (if it was then, the current position should not be INVALID.
         // So if any of those conditions is not true; need to move the mask to the new position.
         def checkInvalid(current: String): F[Option[F[Unit]]] =
-          (epicsSys.maskOffset, epicsSys.lastSelectedMask).mapN { (mo, lsm) =>
-            (!(current === "INVALID" && lsm === mask &&
-              mo =!= 0.0)).option(setMaskIO)
-          }
+          for {
+            mo   <- epicsSys.maskOffset
+            lsm  <- epicsSys.lastSelectedMask
+            check = !(current === "INVALID" && lsm === mask && mo =!= 0.0)
+            _    <-
+              L.debug(
+                s"Check NIFS mask ($check): !(current ($current) === ${'"'}INVALID${'"'} && lsm ($lsm) === mask ($mask) && mo ($mo) =!= 0.0)"
+              )
+          } yield check.option(setMaskIO)
 
         // We need an even smarter set param
         for {
           instMask     <- epicsSys.mask
           setIfInvalid <- checkInvalid(instMask)
-        } yield if (instMask =!= mask) setMaskIO.some else setIfInvalid
```

Comment:
If `instMask =!= mask`, it does not necessarily mean that we need to set the mask. It could be that the right mask is selected, but a mask offset was applied, which is exactly the condition we check inside `checkInvalid()`